lib/kubernetes-deploy/deferred_summary_logging.rb

```diff
@@ -33,6 +33,23 @@ def heading(text, secondary_msg = '', secondary_msg_color = :cyan)
       info(part1 + part2 + part3)
     end
 
+    def record_statuses(resources, operation_name: "deploy")
```

Comment:
It feels kinda wrong to put this in the logger's summary module, since it knows about methods on KubernetesResources. I'm not sure where else to put it though; I'd like the restart and deploy task copies to stay in sync.